deepeval/evaluate.py

```diff
@@ -68,32 +102,108 @@ def create_api_test_case(
         id=test_case.id,
     )
 
+def create_metric_metadata(metric: BaseMetric) -> MetricsMetadata:
+    metadata_kwargs = {
+        'metric': metric.__name__,
+        'score': metric.score,
+        'success': metric.is_successful(),
+        'reason': metric.reason,
+        }
+
+    # Start with required attributes, assuming they must exist in metric
+    for attr in MetricsRequiredParams.values():
+        metadata_kwargs[attr] = getattr(metric, attr)
+    # Then create the optional attrevaluationModelibutes
+    for attr in MetricsOptionalParams.values():
+        if hasattr(metric, attr):
+            metadata_kwargs[attr] = getattr(metric, attr)
+    return MetricsMetadata(**metadata_kwargs)
```

Comment:
Left comment for MetricsMetadata below, might need additional logic to handle the configuration attributes that are not going to Confident